controllers/consoleplugin/consoleplugin_reconciler.go

```diff
@@ -145,3 +161,16 @@ func (r *Reconciler) getObj(ctx context.Context, nsname types.NamespacedName, ob
 	}
 	return obj, nil
 }
+
+func (r *Reconciler) getNamespaceName(desired *flowsv1alpha1.FlowCollectorConsolePlugin) string {
+	if desired.Namespace != "" {
+		return desired.Namespace
+	} else {
+		return r.OperatorNamespace
+	}
+}
+
+func (r *Reconciler) namespaceExist(ctx context.Context, nsName string) (bool, error) {
+	ns, err := r.getObj(ctx, types.NamespacedName{Name: nsName}, &corev1.Namespace{})
```

Comment:
Is there a better way to query a namespace using the controller-runtime client ?

Using a NamespacedName type to query a Namespace feel weird but it works and I couldn't find anything better in the API doc.

Reply:
I don't see anything wrong with that implementation, unless someone else knows any way even better.

Reply:
I don't know, I guess this is fine

Reply:
can't find anything better in the client code, I guess this is the right way :+1: